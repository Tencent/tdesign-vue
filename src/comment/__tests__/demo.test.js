/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/src/comment/_example/base.vue';
import listVue from '@/src/comment/_example/list.vue';
import operationVue from '@/src/comment/_example/operation.vue';
import quoteVue from '@/src/comment/_example/quote.vue';
import replyFormVue from '@/src/comment/_example/reply-form.vue';
import replyVue from '@/src/comment/_example/reply.vue';

const mapper = {
  baseVue,
  listVue,
  operationVue,
  quoteVue,
  replyFormVue,
  replyVue,
};

describe('Comment', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Comment ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
