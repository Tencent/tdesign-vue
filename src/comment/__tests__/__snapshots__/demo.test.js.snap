// Vitest Snapshot v1

exports[`Comment > Comment baseVue demo works fine 1`] = `
<div
  class="t-comment"
>
  <div
    class="t-comment__inner"
  >
    <div
      class="t-comment__avatar"
    >
      <img
        alt=""
        class="t-comment__avatar-image"
        src="https://tdesign.gtimg.com/site/avatar.jpg"
      />
    </div>
    <div
      class="t-comment__content"
    >
      <div
        class="t-comment__author"
      >
        <span
          class="t-comment__name"
        >
          评论作者名
        </span>
        <span
          class="t-comment__time"
        >
          今天16:38
        </span>
      </div>
      <div
        class="t-comment__detail"
      >
        这里是评论者写的评论内容。
      </div>
    </div>
  </div>
</div>
`;

exports[`Comment > Comment listVue demo works fine 1`] = `
<div
  class="t-list t-size-m t-list--split"
  data-v-72dcb149=""
>
  <ul
    class="t-list__inner"
  >
    <li
      class="t-list-item"
      data-v-72dcb149=""
    >
      <div
        class="t-list-item-main"
      >
        <div
          class="t-list-item__content"
        >
          <div
            class="t-comment"
            data-v-72dcb149=""
          >
            <div
              class="t-comment__inner"
            >
              <div
                class="t-comment__avatar"
              >
                <img
                  alt=""
                  class="t-comment__avatar-image"
                  src="https://tdesign.gtimg.com/site/avatar.jpg"
                />
              </div>
              <div
                class="t-comment__content"
              >
                <div
                  class="t-comment__author"
                >
                  <span
                    class="t-comment__name"
                  >
                    评论作者名A
                  </span>
                  <span
                    class="t-comment__time"
                  >
                    今天16:38
                  </span>
                </div>
                <div
                  class="t-comment__detail"
                >
                  评论作者名A写的评论内容。
                </div>
                <ul
                  class="t-comment__actions"
                >
                  <li>
                    <span
                      data-v-72dcb149=""
                    >
                      <svg
                        class="t-icon t-icon-thumb-up"
                        data-v-72dcb149=""
                      >
                        <use
                          href="#t-icon-thumb-up"
                        />
                      </svg>
                      <span
                        class="action-text"
                        data-v-72dcb149=""
                      >
                        6
                      </span>
                    </span>
                  </li>
                  <li>
                    <span
                      data-v-72dcb149=""
                    >
                      <svg
                        class="t-icon t-icon-chat"
                        data-v-72dcb149=""
                      >
                        <use
                          href="#t-icon-chat"
                        />
                      </svg>
                      <span
                        class="action-text"
                        data-v-72dcb149=""
                      >
                        回复
                      </span>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="t-list-item"
      data-v-72dcb149=""
    >
      <div
        class="t-list-item-main"
      >
        <div
          class="t-list-item__content"
        >
          <div
            class="t-comment"
            data-v-72dcb149=""
          >
            <div
              class="t-comment__inner"
            >
              <div
                class="t-comment__avatar"
              >
                <img
                  alt=""
                  class="t-comment__avatar-image"
                  src="https://tdesign.gtimg.com/site/avatar.jpg"
                />
              </div>
              <div
                class="t-comment__content"
              >
                <div
                  class="t-comment__author"
                >
                  <span
                    class="t-comment__name"
                  >
                    评论作者名B
                  </span>
                  <span
                    class="t-comment__time"
                  >
                    今天16:38
                  </span>
                </div>
                <div
                  class="t-comment__detail"
                >
                  评论作者名B写的评论内容。
                </div>
                <ul
                  class="t-comment__actions"
                >
                  <li>
                    <span
                      data-v-72dcb149=""
                    >
                      <svg
                        class="t-icon t-icon-thumb-up"
                        data-v-72dcb149=""
                      >
                        <use
                          href="#t-icon-thumb-up"
                        />
                      </svg>
                      <span
                        class="action-text"
                        data-v-72dcb149=""
                      >
                        6
                      </span>
                    </span>
                  </li>
                  <li>
                    <span
                      data-v-72dcb149=""
                    >
                      <svg
                        class="t-icon t-icon-chat"
                        data-v-72dcb149=""
                      >
                        <use
                          href="#t-icon-chat"
                        />
                      </svg>
                      <span
                        class="action-text"
                        data-v-72dcb149=""
                      >
                        回复
                      </span>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="t-list-item"
      data-v-72dcb149=""
    >
      <div
        class="t-list-item-main"
      >
        <div
          class="t-list-item__content"
        >
          <div
            class="t-comment"
            data-v-72dcb149=""
          >
            <div
              class="t-comment__inner"
            >
              <div
                class="t-comment__avatar"
              >
                <img
                  alt=""
                  class="t-comment__avatar-image"
                  src="https://tdesign.gtimg.com/site/avatar.jpg"
                />
              </div>
              <div
                class="t-comment__content"
              >
                <div
                  class="t-comment__author"
                >
                  <span
                    class="t-comment__name"
                  >
                    评论作者名C
                  </span>
                  <span
                    class="t-comment__time"
                  >
                    今天16:38
                  </span>
                </div>
                <div
                  class="t-comment__detail"
                >
                  评论作者名C写的评论内容。
                </div>
                <ul
                  class="t-comment__actions"
                >
                  <li>
                    <span
                      data-v-72dcb149=""
                    >
                      <svg
                        class="t-icon t-icon-thumb-up"
                        data-v-72dcb149=""
                      >
                        <use
                          href="#t-icon-thumb-up"
                        />
                      </svg>
                      <span
                        class="action-text"
                        data-v-72dcb149=""
                      >
                        6
                      </span>
                    </span>
                  </li>
                  <li>
                    <span
                      data-v-72dcb149=""
                    >
                      <svg
                        class="t-icon t-icon-chat"
                        data-v-72dcb149=""
                      >
                        <use
                          href="#t-icon-chat"
                        />
                      </svg>
                      <span
                        class="action-text"
                        data-v-72dcb149=""
                      >
                        回复
                      </span>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
  <div
    class="t-list__load"
  />
</div>
`;

exports[`Comment > Comment operationVue demo works fine 1`] = `
<div
  class="t-comment"
  data-v-70d43968=""
>
  <div
    class="t-comment__inner"
  >
    <div
      class="t-comment__avatar"
    >
      <img
        alt=""
        class="t-comment__avatar-image"
        src="https://tdesign.gtimg.com/site/avatar.jpg"
      />
    </div>
    <div
      class="t-comment__content"
    >
      <div
        class="t-comment__author"
      >
        <span
          class="t-comment__name"
        >
          评论作者名
        </span>
        <span
          class="t-comment__time"
        >
          今天16:38
        </span>
      </div>
      <div
        class="t-comment__detail"
      >
        这里是评论者写的评论内容。
      </div>
      <ul
        class="t-comment__actions"
      >
        <li>
          <span
            data-v-70d43968=""
          >
            <svg
              class="t-icon t-icon-thumb-up"
              data-v-70d43968=""
            >
              <use
                href="#t-icon-thumb-up"
              />
            </svg>
            <span
              class="action-text"
              data-v-70d43968=""
            >
              6
            </span>
          </span>
        </li>
        <li>
          <span
            data-v-70d43968=""
          >
            <svg
              class="t-icon t-icon-chat"
              data-v-70d43968=""
            >
              <use
                href="#t-icon-chat"
              />
            </svg>
            <span
              class="action-text"
              data-v-70d43968=""
            >
              回复
            </span>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Comment > Comment quoteVue demo works fine 1`] = `
<div
  class="t-comment"
  data-v-4829af3e=""
>
  <div
    class="t-comment__inner"
  >
    <div
      class="t-comment__avatar"
    >
      <img
        alt=""
        class="t-comment__avatar-image"
        src="https://tdesign.gtimg.com/site/avatar.jpg"
      />
    </div>
    <div
      class="t-comment__content"
    >
      <div
        class="t-comment__author"
      >
        <span
          class="t-comment__name"
        >
          评论作者名
        </span>
        <span
          class="t-comment__time"
        >
          今天16:38
        </span>
      </div>
      <div
        class="t-comment__detail"
      >
        这里是评论者写的评论内容。
      </div>
      <div
        class="t-comment__quote"
      >
        <div
          class="t-comment"
          data-v-4829af3e=""
        >
          <div
            class="t-comment__inner"
          >
            <div
              class="t-comment__content"
            >
              <div
                class="t-comment__author"
              >
                <span
                  class="t-comment__name"
                >
                  引用内容标题
                </span>
              </div>
              <div
                class="t-comment__detail"
              >
                引用评论内容。
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        class="t-comment__actions"
      >
        <li>
          <span
            data-v-4829af3e=""
          >
            <svg
              class="t-icon t-icon-thumb-up"
              data-v-4829af3e=""
            >
              <use
                href="#t-icon-thumb-up"
              />
            </svg>
            <span
              class="action-text"
              data-v-4829af3e=""
            >
              6
            </span>
          </span>
        </li>
        <li>
          <span
            data-v-4829af3e=""
          >
            <svg
              class="t-icon t-icon-chat"
              data-v-4829af3e=""
            >
              <use
                href="#t-icon-chat"
              />
            </svg>
            <span
              class="action-text"
              data-v-4829af3e=""
            >
              回复
            </span>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Comment > Comment replyFormVue demo works fine 1`] = `
<div
  class="t-comment"
  data-v-02d347b2=""
>
  <div
    class="t-comment__inner"
  >
    <div
      class="t-comment__avatar"
    >
      <img
        alt=""
        class="t-comment__avatar-image"
        src="https://tdesign.gtimg.com/site/avatar.jpg"
      />
    </div>
    <div
      class="t-comment__content"
    >
      <div
        class="t-comment__detail"
      >
        <div
          class="form-container"
          data-v-02d347b2=""
        >
          <div
            class="t-textarea"
            data-v-02d347b2=""
          >
            <textarea
              class="t-textarea__inner"
              placeholder="请输入内容"
              unselectable="off"
            />
          </div>
          <button
            class="form-submit t-button t-size-m t-button--variant-base t-button--theme-primary"
            data-v-02d347b2=""
            type="button"
          >
            <span
              class="t-button__text"
            >
              回复
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Comment > Comment replyVue demo works fine 1`] = `
<div
  class="t-comment"
  data-v-1a7615cf=""
>
  <div
    class="t-comment__inner"
  >
    <div
      class="t-comment__avatar"
    >
      <img
        alt=""
        class="t-comment__avatar-image"
        src="https://tdesign.gtimg.com/site/avatar.jpg"
      />
    </div>
    <div
      class="t-comment__content"
    >
      <div
        class="t-comment__author"
      >
        <span
          class="t-comment__name"
        >
          评论作者名
        </span>
        <span
          class="t-comment__time"
        >
          今天16:38
        </span>
      </div>
      <div
        class="t-comment__detail"
      >
        这里是评论者写的评论内容。
      </div>
      <ul
        class="t-comment__actions"
      >
        <li>
          <span
            data-v-1a7615cf=""
          >
            <svg
              class="t-icon t-icon-thumb-up"
              data-v-1a7615cf=""
            >
              <use
                href="#t-icon-thumb-up"
              />
            </svg>
            <span
              class="action-text"
              data-v-1a7615cf=""
            >
              6
            </span>
          </span>
        </li>
        <li>
          <span
            data-v-1a7615cf=""
          >
            <svg
              class="t-icon t-icon-chat"
              data-v-1a7615cf=""
            >
              <use
                href="#t-icon-chat"
              />
            </svg>
            <span
              class="action-text"
              data-v-1a7615cf=""
            >
              回复
            </span>
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="t-comment__reply"
  >
    <div
      class="t-comment"
      data-v-1a7615cf=""
    >
      <div
        class="t-comment__inner"
      >
        <div
          class="t-comment__avatar"
        >
          <img
            alt=""
            class="t-comment__avatar-image"
            src="https://tdesign.gtimg.com/site/avatar.jpg"
          />
        </div>
        <div
          class="t-comment__content"
        >
          <div
            class="t-comment__author"
          >
            <span
              class="t-comment__name"
            >
              <span
                data-v-1a7615cf=""
              >
                评论作者名B
              </span>
              <svg
                class="author-icon t-icon t-icon-caret-right-small t-size-s"
                data-v-1a7615cf=""
              >
                <use
                  href="#t-icon-caret-right-small"
                />
              </svg>
              <span
                data-v-1a7615cf=""
              >
                评论作者名A
              </span>
            </span>
            <span
              class="t-comment__time"
            >
              今天16:38
            </span>
          </div>
          <div
            class="t-comment__detail"
          >
            这里是评论者写的评论内容。
          </div>
          <ul
            class="t-comment__actions"
          >
            <li>
              <span
                data-v-1a7615cf=""
              >
                <svg
                  class="t-icon t-icon-thumb-up"
                  data-v-1a7615cf=""
                >
                  <use
                    href="#t-icon-thumb-up"
                  />
                </svg>
                <span
                  class="action-text"
                  data-v-1a7615cf=""
                >
                  6
                </span>
              </span>
            </li>
            <li>
              <span
                data-v-1a7615cf=""
              >
                <svg
                  class="t-icon t-icon-chat"
                  data-v-1a7615cf=""
                >
                  <use
                    href="#t-icon-chat"
                  />
                </svg>
                <span
                  class="action-text"
                  data-v-1a7615cf=""
                >
                  回复
                </span>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
